lab5.py

```diff
@@ -0,0 +1,92 @@
+from enum import Enum
+import random
+from datetime import datetime
+
+class Movie:
+    '''
+    this class describes objects of type "Movie" using certain fields
+    '''
+    def __init__(self, id, title, ranking, character_number,
+                 ticket_price, genre, comment ,year, month, day):
+        self.id = id
+        self.title = title
+        self.ranking = ranking
+        self.character_number = character_number
+        self.ticket_price = ticket_price
+        self.comment = comment
+        self.__genre = genre
+        self.__date = datetime(year, month, day)
+        
+    @property
+    def genre(self):
+        return self.__genre
+
+    @genre.setter
+    def genre(self, genre):
+        self.__genre = genre
+    @property
+    def date(self):
+        return self.__date
+
+    @date.setter
+    def date_(self, date):
+        self.__date = date    
+
+class Genre(Enum):
+    '''
+    the Genre class is used to define a enumerate of movie genres using enumerations
+    '''
+    ACTION = 1
+    COMEDY = 2
+    DRAMA = 3
+    FANTASY = 4
+
+class Cinema:
+    '''
+    this class describes objects of type "Cinema" using certain fields
+    '''
+    def __init__(self, name, location,  *movies):
+        self.name = name
+        self.location = location
+        self.movies = list(movies)
+       
+    @property
+    def cinema_date(self):
+        return self._date
+
+    @cinema_date.setter
+    def cinema_date(self, date):
+        self._date = date
+
+    def calculate_profit(self, day):
+        '''
+        this method calculates the profit for
+        the day based on the number of tickets purchased
+        '''
+        for movie in self.movies:
+            all_profit = random.randint(1,1000)
+            if movie.date == day:
+                sold_tickets = random.randint(1,1000) 
+                all_profit =+( sold_tickets * movie.ticket_price)
+            return all_profit
+
+    def sort_by_date(self):
+        '''
+        this method sorts movies from newest to oldest
+        '''
+        self.movies.sort(key=lambda x: x.date, reverse=True)
+        
+if __name__ == '__main__':
+    movie1 = Movie(123456789, "fiting club", 1, 15, 120, Genre.ACTION,
+                   "перше правило", 2006, 6, 3)
+    movie2 = Movie(123456789, "Godfather ", 1, 15, 130, Genre.DRAMA,
+                   " де моє кільце , Джони", 2004, 6, 3)
+    movie3 = Movie(123456789, "spider man ", 1, 15, 10, Genre.FANTASY,
+                   "Я вибачаю тебе", 2002, 6, 3)
```

Comment:
please give meaningful names for movie variable name - fiting_club, godfather, etc

Reply:
I give  meaningful names for all movie variable name